### PR TITLE
[DROOLS-3743] Implement Test scenario coverage report for RULE

### PR DESCRIPTION
**Pull request by danielezonca**
This PR contains backend support and UI changes to calculate test coverage KPIs for rules

NOTE: now if a rule has been executed multiple times in the same scenario, the last section of the report (rules executed for each scenario) will show also this number. I.e. `com.rule (3)`

@kkufova @Rikkola @yesamer 

PRs list:
- https://github.com/kiegroup/droolsjbpm-knowledge/pull/396
- https://github.com/kiegroup/drools/pull/2471
- https://github.com/kiegroup/drools-wb/pull/1197

### CODE REVIEW

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by danielezonca** [comment]
jenkins retest this please

**Comment by kkufova** [comment]
- if an agenda group is applied, the number of available rules is calculated accordingly;
- if an agenda group is selected, it will execute that group of rules + all the rules without any agenda group defined (= the defined behavior of agenda group)

**Comment by danielezonca** [comment]
@kkufova 
[reference](https://docs.jboss.org/drools/release/7.24.0.Final/drools-docs/html_single/index.html#_agenda_groups_for_rules)

**Comment by kkufova** [comment]
@danielezonca, I have only two comments:

1. I cannot reproduce this, but when I opened a fresh empty test scenario asset and checked the coverage report, the percentage of rules fired was 00.00%. However, now, after working with Business Central for a while, the coverage report looks like this:



You can see that the percentage now is NaN%. I think using 0% or 00.00% is better in both cases: when the table is completely empty or when the agenda groups are incompatible. I am completely sure that I saw 00.00%, but I cannot get to this number anymore.

2. I think the package mark should be changed from the format _com.name of the rule_ to _name of the rule (com)_.

**Comment by danielezonca** [comment]
@kkufova 
Done
1. now if there are no rules/decisions it displays a proper message instead of a broken report
`Coverage report is not available when there are no decisions`
`Coverage report is not available when there are no rules`
2. done. Please verify with DRL in a package and with DRL in root folder

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/resources/org/drools/workbench/screens/scenariosimulation/client/resources/i18n/ScenarioSimulationEditorConstants.properties [hunk 1]
```diff
@@ -157,3 +157,10 @@ apply=Apply
 errorPopoverMessageFailedWithError=The expected value is ''{0}'' but the actual one is ''{1}''.
 errorPopoverMessageFailedWithException=Error message: {0}
 dateFormatPlaceholder=Insert date as yyyy-mm-dd
+rulesFired=Rules fired for test scenario
+numberOfTimesRulesFired=A number of times each rule has been fired:
+reportAvailableRuleLabel=Available rules:
+reportExecutedRuleLabel=Fired rules:
+reportCoverageRuleLabel=Percentage of rules fired:
+noDecisionsAvailable=Coverage report is not available when there are no decisions
```
[the remark this answers is not in the document]
Comment (danielezonca) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/resources/org/drools/workbench/screens/scenariosimulation/client/resources/i18n/ScenarioSimulationEditorConstants.properties]: Done

File: drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/resources/org/drools/workbench/screens/scenariosimulation/client/resources/i18n/ScenarioSimulationEditorConstants.properties [hunk 2]
```diff
@@ -157,3 +157,10 @@ apply=Apply
...
+noRulesAvailable=Coverage report is not available when there are no rules
```
[the remark this answers is not in the document]
Comment (danielezonca) [reply drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/resources/org/drools/workbench/screens/scenariosimulation/client/resources/i18n/ScenarioSimulationEditorConstants.properties]: Done

**Pull request merged**